src/payment/strategies/checkoutcom-custom/checkoutcom-ideal/checkoutcom-ideal-payment-strategy.ts

```diff
@@ -0,0 +1,46 @@
+import { InternalCheckoutSelectors } from '../../../../checkout';
+import { OrderRequestBody } from '../../../../order';
+import { PaymentArgumentInvalidError } from '../../../errors';
+import { PaymentInstrument, WithCheckoutcomiDealInstrument } from '../../../payment';
+import { PaymentRequestOptions } from '../../../payment-request-options';
+import CheckoutcomCustomPaymentStrategy from '../checkoutcom-custom-payment-strategy';
+
+const CHECKOUTCOM_IDEAL_PAYMENT_METHOD = 'ideal';
+export default class CheckoutcomiDealPaymentStrategy extends CheckoutcomCustomPaymentStrategy {
+
+    protected async _executeWithoutHostedForm(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment, ...order } = payload;
+        const paymentData = payment?.paymentData;
+
+        if (!payment || !paymentData) {
```

Comment:
`if (!paymentData)` should suffice

Reply:
it should, but the linter requires `payment` to be validated before using it.

Reply:
right I see, I can see you are using it further down